tests/vgg_tests.cpp

```diff
@@ -0,0 +1,507 @@
+/**
+ * @file tests/vgg_tests.cpp
+ * @author Shubham Agrawal
+ *
+ * Tests the VGG models.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include "../models/vgg/vgg.hpp"
+
+#include "test_catch_tools.hpp"
+#include "catch.hpp"
+#include "serialization.hpp"
+
+using namespace mlpack;
+
+/**
+ * Checks for the output dimensions of the model.
+ *
+ * @tparam ModelType Type of model to check.
+ *
+ * @param model The model to test.
+ * @param input Input to pass to the model.
+ * @param n_rows Output rows to check against.
+ * @param n_cols Output columns to check against..
+ */
+template <typename ModelType>
+void ModelDimTest(ModelType& model,
+               arma::mat& input,
+               const size_t n_rows = 1000,
+               const size_t n_cols = 1)
+{
+  arma::mat output;
+  model.Predict(input, output);
+  REQUIRE(output.n_rows == n_rows);
+  REQUIRE(output.n_cols == n_cols);
+}
+
+/**
+ * Checks for the output sum of the model for a 
+ *     single and multiple batch input.
+ *
+ * @tparam ModelType Type of model to check.
+ *
+ * @param model The model to test.
+ * @param input Input to pass to the model.
+ * @param singleBatchOutput Sum of the output of a single batch.
+ * @param multipleBatchOutput Sum of the output of Multiple batches.
+ * @param numBatches Number of batches to create for input.
+ */
+template <typename ModelType>
+void PreTrainedModelTest(ModelType& model,
+                         arma::mat& input,
+                         const double singleBatchOutput,
+                         const double multipleBatchOutput,
+                         const size_t numBatches = 4)
+{
+  arma::mat multipleBatchInput(input.n_rows, numBatches), output;
+  input.ones();
+  multipleBatchInput.ones();
+
+  // Run prediction for single batch.
+  model.Predict(input, output);
+  REQUIRE(arma::accu(output) == Approx(singleBatchOutput).epsilon(1e-2));
+
+  // Run prediction for multiple batch.
+  model.Predict(multipleBatchInput, output);
+  REQUIRE(arma::accu(output) == Approx(multipleBatchOutput).epsilon(1e-2));
+}
+
+// General ANN serialization test.
+template<typename LayerType>
+void ModelSerializationTest(LayerType& layer)
+{
+  arma::mat input(224 * 224 * 3, 10, arma::fill::randu);
+  arma::mat output(1000, 10, arma::fill::randu);
+
+  ann::FFN<> model;
+  model.Add<LayerType>(layer);
+
+	model.InputDimensions() = std::vector<size_t>({224, 224, 3});
```

Comment:
Looks like there are some style issues, with tab characters appearing here and in the later tests.  Can you fix those?